### PR TITLE
Update ROC target documentation.

### PR DESCRIPTION
**Pull request by stuartarchibald**
This updates the ROC target documentation to reflect the work on
dGPUs and retires the work on APU. It:

* Accommodates the HSA->ROC changes
* Adds instructions on installation and setup
* Updates examples and makes sure they work!
* Adds new device function example
* Adds section on ROC ufuncs, including async streaming gufuncs

Minor code changes:

* numba.roc.agents now works and lists the available agents

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Needs https://github.com/numba/numba/pull/3176 and https://github.com/numba/numba/pull/3177 for the actual code examples to work!

**Comment by seibert** [comment]
Now that #3176 and #3177 are merged, refreshing Travis CI to see if docs build.

File: setup.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def rec(path, pkg_name):
         "numba.annotations": ["*.html"],
         # Various test data
         "numba.cuda.tests.cudadrv.data": ["*.ptx"],
-        "numba.hsa.tests.hsadrv": ["*.brig"],
+        "numba.roc.tests.hsadrv": ["*.brig"],
```
Comment (seibert) [inline setup.py]: I don't think we are still shipping any .brig files.  There is one unit test, but it references a file under `/opt/rocm'.

**Comment by seibert** [comment]
Looks great!  I had one minor nit about setup.py.

File: setup.py (hunk 1 above)
Comment (stuartarchibald) [reply setup.py]: Ah yes, thanks, I don't think any brigs are shipped now.

**Comment by stuartarchibald** [comment]
Thanks for the review, I've addressed https://github.com/numba/numba/pull/3178#discussion_r206661804 in 95939f1.

**Pull request merged**